realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -579,11 +579,8 @@ public long getPrimaryKey() {
                 return NO_PRIMARY_KEY; // Free table = No primary key
             }
 
-            String tableName = getName();
-            if (tableName.startsWith(TABLE_PREFIX)) {
-                tableName = tableName.substring(TABLE_PREFIX.length());
-            }
-            long rowIndex = pkTable.findFirstString(PRIMARY_KEY_CLASS_COLUMN_INDEX, tableName);
+            String className = tableNameToClassName(getName());
+            long rowIndex = pkTable.findFirstString(PRIMARY_KEY_CLASS_COLUMN_INDEX, className);
```

Comment:
@zaki50 That only impacts on pkTable, right? So it is not necessary to be in `tableNameToClassName`